### PR TITLE
Mysadnaangular8 - Draft with Firebase

### PR DESCRIPTION
**Pull request by a3l5o7n9**
Includes the first try of incorporating Firebase and Protoc.

### CODE REVIEW

File: projects/mysadna/src/app/app.component.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { Component } from '@angular/core';
 
 import { EncodingService } from './services/encoding.service';
 import { FirebaseService } from './services/firbase.service';
-import { Data } from './proto';
+import { Data } from '@/proto';
```
Comment (webdevelopland) [inline projects/mysadna/src/app/app.component.ts]: Looks like your project doesn't have `fs-extra`, which is needed to create proto function.
How to fix it:
```
npm i fs-extra --save-dev
```

File: projects/mysadna/src/app/components/project-details/project-details.component.ts [hunk 2]
```diff
@@ -2,9 +2,9 @@ import { Component, OnInit } from '@angular/core';
 import { ActivatedRoute } from '@angular/router';
 import { Location } from '@angular/common';
 
-import { User } from '../../models/user';
-import { Project } from '../../models/project';
-import { DataService } from '../../services/data.service';
+import { User } from '@/models/user';
```
Comment (webdevelopland) [inline projects/mysadna/src/app/components/project-details/project-details.component.ts]: :+1:

File: projects/mysadna/src/app/app.component.ts [hunk 3]
```diff
@@ -1,10 +1,25 @@
...
+
+
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
   styleUrls: ['./app.component.scss']
 })
 export class AppComponent {
-  title = 'mysadna2';
+  title = 'mysadna';
```
[the remark this answers is not in the document]
Comment (a3l5o7n9) [reply projects/mysadna/src/app/app.component.ts]: Removed.

File: projects/mysadna/src/app/components/project-details/project-details.component.ts [hunk 4]
```diff
@@ -2,9 +2,9 @@ import { Component, OnInit } from '@angular/core';
 import { ActivatedRoute } from '@angular/router';
 import { Location } from '@angular/common';
 
-import { User } from '../models/user';
-import { Project } from '../models/project';
-import { DataService } from '../services/data.service';
+import { User } from '../../models/user';
```
[the remark this answers is not in the document]
Comment (a3l5o7n9) [reply projects/mysadna/src/app/components/project-details/project-details.component.ts]: Done.

File: projects/mysadna/src/app/proto/functions/data_pb.d.ts [hunk 5]
```diff
@@ -0,0 +1,177 @@
+// package: com.hasadna.mysadna
```
[the remark this answers is not in the document]
Comment (a3l5o7n9) [reply projects/mysadna/src/app/proto/functions/data_pb.d.ts]: Obviously, I've managed to push new commits again.

File: projects/mysadna/src/app/app.component.scss [hunk 6]
```diff
@@ -1,3 +1,8 @@
+.app-container {
+  width: 100%;
+  height: 100%;
```
[the remark this answers is not in the document]
Comment (a3l5o7n9) [reply projects/mysadna/src/app/app.component.scss]: What about width?

File: projects/mysadna/src/app/app.component.ts (hunk 1 above)
Comment (a3l5o7n9) [reply projects/mysadna/src/app/app.component.ts]: installed.

File: projects/mysadna/src/app/app.component.scss (hunk 6 above)
[the remark this answers is not in the document]
Comment (webdevelopland) [reply projects/mysadna/src/app/app.component.scss]: `width: 100%` is ok

File: projects/mysadna/src/app/components/project-details/project-details.component.scss [hunk 7]
```diff
@@ -1,7 +1,7 @@
 .project-details {
   text-align: center;
   width: 100%;
-  height: 100%;
```
Comment (webdevelopland) [inline projects/mysadna/src/app/components/project-details/project-details.component.scss]: Thanks for doing it for all instances :+1:

**Pull request merged**